### PR TITLE
Split the function of "PLAY COS".

### PR DESCRIPTION
**Pull request by Misaliet**
The original "PLAY COS" is divided into three functions: rendering training set ("PLAY COS"), rendering test set ("PLAY COS TEST"), and packaging ("ZIP").

### CODE REVIEW

**Comment by Misaliet** [comment]
Hi, I really like your code for generating training data for NeRF.
I made a simple modification according to my needs. I found frameworks like Instant-NGP will raise errors directly if there is a "transforms_test.json" file but without corresponding test images. So I want to add a function to render test images as well.
I don't know much about the development of the Blender plug-in, but I found that the asynchronous operation of Render will cause some bugs if we do render operation twice inside one function. So I separated this into there functions.
My solution is far from perfect, I understand if you don't accept my pull request.
Later, I plan to add the function of rendering the corresponding depth map and normal map at the same time.
If my code isn't good enough for you, I'll keep it in my own repository.

**Comment by maximeraafat** [comment]
Hi @Misaliet, glad you like the code!

Thanks for the suggestions and your contribution. I had the opportunity to look at your changes and understand what you are trying to achieve. However, Blender actually provides test images for free by simply selecting the camera and rendering the frames (see #13 and feel free to contribute to the discussion). I do not think there really is a need for additional functionalities to do so.

As for the quality of your edits, I am far from an expert, but in order to keep the code short and clean I would have kept everything within a single class. You could then differentiate whether to render training or testing frames via a toggle which would flip a boolean property shared across the whole add-on.

Finally, **BlenderNeRF** actually just renders whatever the compositor will have as output. You can customise it in the following way to get additional normals and depths without much effort. Under the **ViewLayer Properties** panel, check the **Z** and the **Normals** boxes as in the following screenshot.

viewlayer

Then, replicate the following node setup in the compositor by simply replacing the first part of the path (here `/Users/maxime/Downloads/`) with the **Save Path** from **BlenderNeRF**, and the second last part (here `dataset`) with the name of your dataset.

depth_and_normals

This will directly save the normals and depths for the corresponding training frames within their respective folders inside the final zipped dataset.

I hope this helped and I wish you luck with your projects! If you have any further suggestions or questions, feel free to continue here or open a new issue :)
